gatherers/DockerComposeGatherer.go

```diff
@@ -0,0 +1,70 @@
+package gatherers
+
+import (
+	"github.com/bomoko/lagoon-facts/utils"
+	"log"
+)
+
+type dockerComposeGatherer struct {
+	GatheredFacts []GatheredFact
+}
+
+type dockerComposeService struct {
+	Name string
+	Description string
+	Image string
+	Labels map[interface {}]interface {}
+	Build map[interface {}]interface {}
+}
+
+func (p *dockerComposeGatherer) AppliesToEnvironment() bool {
+	applies := false
+
+	lagoonConfigBytestream, err := LoadDockerComposeConfig("docker-compose.yml")
+	if err != nil {
+		log.Printf("Couldn't load docker-compose.yml file: ", err.Error())
+	}
+
+	config, err := utils.UnmarshallYamlToStructure(lagoonConfigBytestream)
+	if err != nil {
+		log.Fatalf("There was an issue unmarshalling the docker-compose.yml file: %s", err)
+	}
+
+	if config.Services != nil {
+		applies = true
+
+		for k, element := range config.Services {
+			serviceItems := element.(map[interface {}]interface {})
+
+			var service dockerComposeService
+			service.Name = k
+			service.Description = "Service found in docker-compose.yml"
+
+			for j, val := range serviceItems {
+				if str, ok := val.(string); ok {
+					if j == "image" {
+						service.Image = str
```

Comment:
Additionally, collecting the service.Image will be overridden for DBaaS-type services, which won't be apparent here?